### ISSUE TITLE
[firebase_database] java.lang.Integer cannot be cast to java.lang.Long on calling FirebaseDatabase.setPersistenceCacheSizeBytes on android

### ISSUE THREAD

**Issue by animeshjain** [FirebaseExtended/flutterfire#81]
in `firebase_database v3.0.7` on calling   `FirebaseDatabase.instance.setPersistenceCacheSizeBytes(10000000);` I get the following error on Android

```
E/MethodChannel#plugins.flutter.io/firebase_database(30947): Failed to handle method call
E/MethodChannel#plugins.flutter.io/firebase_database(30947): java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at io.flutter.plugins.firebase.database.FirebaseDatabasePlugin.onMethodCall(FirebaseDatabasePlugin.java:284)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:222)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:96)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:643)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at android.os.MessageQueue.next(MessageQueue.java:323)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at android.os.Looper.loop(Looper.java:135)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at android.app.ActivityThread.main(ActivityThread.java:5461)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
E/MethodChannel#plugins.flutter.io/firebase_database(30947): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
E/flutter (30947): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(error, java.lang.Integer cannot be cast to java.lang.Long, null)
E/flutter (30947): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:564:7)
E/flutter (30947): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:316:33)
E/flutter (30947): <asynchronous suspension>
E/flutter (30947): #2      FirebaseDatabase.setPersistenceCacheSizeBytes (package:firebase_database/src/firebase_database.dart:120:40)
E/flutter (30947): <asynchronous suspension>
```

**Comment by jpeiffer** [comment]
I get the same problem and had to revert to 3.0.6 or earlier.

Looks like the relevant change happened here:
https://github.com/FirebaseExtended/flutterfire/commit/df54899f6144b00e81229fca33f439242b858454#diff-75d8ced9bfc6b037b8cbc18da165fcd5L282

The previous code correctly pulled it out as an `Integer` and then it was widened into a `long` primitive.  The updated code removes the autoboxing / unboxing warning, but tries to take the same `Integer` object and cast it to a `Long`.  Primitives can widen automatically like that (ie: a `long` can be assigned an `int` value), but object cannot.

This looks to be a pretty simple fix.  I went ahead and created a PR for it:
https://github.com/FirebaseExtended/flutterfire/pull/85

**Comment by alexZaicev** [comment]
I have experience the same issue in my App as well when trying to set cache bytes
```
E/flutter ( 4571): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(error, java.lang.Integer cannot be cast to java.lang.Long, null)
E/flutter ( 4571): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:569:7)
E/flutter ( 4571): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:316:33)
E/flutter ( 4571): <asynchronous suspension>
E/flutter ( 4571): #2      FirebaseDatabase.setPersistenceCacheSizeBytes (package:firebase_database/src/firebase_database.dart:120:40)
```

**Comment by solokingjec** [comment]
I have the same issue here.. Reverting to an old version doesn't help either.

<img width="2048" alt="Screenshot 2019-10-07 at 12 53 01" src="https://user-images.githubusercontent.com/6312641/66302511-bd333680-e901-11e9-8ef0-95ee53e6d02d.png">
<img width="2048" alt="Screenshot 2019-10-07 at 12 54 59" src="https://user-images.githubusercontent.com/6312641/66302513-bdcbcd00-e901-11e9-8412-4e4fdd15cf38.png">

**Comment by animeshjain** [comment]
@solokingjec Try specifying the exact version 

```
firebase_database: 3.0.6
```
 instead of something like

```
firebase_database: ^3.0.6
```

in your `pubspec.yaml`

Then run flutter pub get again to go back to the older version. Something like `^3.0.4` will not revert back to an earlier version if there is a higher minor version available.

**Comment by alexZaicev** [comment]
When this fix will be merged and published? It seems an easy to fix error, and somebody already submitted a pull request..

**Comment by abaker2010** [comment]
Is there any news on this fix? The version of Firebase Database that I am using is 3.1.0. The weird thing about this is that I am only getting the error about the persistence when running on Android but not on the iPhone. 
I am trying to set the persistence to be at 10MB with this: 


`
      _firebaseDatabase.setPersistenceEnabled(true);
      _firebaseDatabase.setPersistenceCacheSizeBytes(10000000);`

I have been getting this error: 

`E/flutter (32212): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(error, java.lang.Integer cannot be cast to java.lang.Long, null)
E/flutter (32212): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:569:7)
E/flutter (32212): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:316:33)
E/flutter (32212): <asynchronous suspension>
E/flutter (32212): #2      FirebaseDatabase.setPersistenceCacheSizeBytes (package:firebase_database/src/firebase_database.dart:120:40)`

**Comment by rignaneseleo** [comment]
Same issue. Using the version 3.1.1

**Comment by abaker2010** [comment]
Any new on a potential work around?

**Comment by jpeiffer** [comment]
Created a new PR based on the latest code: https://github.com/FirebaseExtended/flutterfire/pull/1708

In the mean time, you can work around the issue by pointing to the fix hosted in my repo:
```
  firebase_database:
    git:
      url: https://github.com/jpeiffer/flutterfire
      path: packages/firebase_database
      ref: 369ac426ce8a28ce1720643d5c69614dc070ec8a
```

**Comment by HaroldBeyer** [comment]
any news?

**Comment by Izzur** [comment]
While waiting @jpeiffer fix being merged I try to use number bigger than `java.Lang.integer` (2,147,483,647) and its not throwing any error  
  
```
db.setPersistenceCacheSizeBytes(2147483648);
```
  
Doing so will set persistent cache to 2GB

**Comment by davidcantonnoteges** [comment]
Hello friends, any news?

**Comment by Ehesp** [comment]
This has been merged - will release later on once access has been sorted.

**Comment by Ehesp** [comment]
Fix available in 3.1.5
